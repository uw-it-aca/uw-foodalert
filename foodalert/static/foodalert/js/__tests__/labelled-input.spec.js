"use strict";
import Vue from 'vue';
import BootstrapVue from 'bootstrap-vue'
Vue.use(BootstrapVue);

import {mount} from '@vue/test-utils';
import LabelledInput from '../components/labelled-input.vue';

describe('Labelled Input', function () {
    describe('renders correctly', function () {
        test('textarea', function () {
            var wrapper = mount(LabelledInput, {
                propsData: {
                    labelText: "Label Text",
                    exampleText: "Example Text",
                    warningText: "Warning Text",
                }
            });
            expect(wrapper.element).toMatchSnapshot();
        });

        test('checkbox', function () {
            var wrapper = mount(LabelledInput, {
                propsData: {
                    type: "checkbox",
                    labelText: "Label Text",
                    exampleText: "Example Text",
                    warningText: "Warning Text",
                }
            });
            expect(wrapper.element).toMatchSnapshot();
        })

        test('time', function () {
            var wrapper = mount(LabelledInput, {
                propsData: {
                    type: "time",
                    labelText: "Label Text",
                    exampleText: "Example Text",
                    warningText: "Warning Text",
                }
            });
            expect(wrapper.element).toMatchSnapshot();
        })

        test('buttons', function () {
            var wrapper = mount(LabelledInput, {
                propsData: {
                    type: "buttons",
                    labelText: "Label Text",
                    exampleText: "Example Text",
                    warningText: "Warning Text",
                }
            });
            expect(wrapper.element).toMatchSnapshot();
        })
    });
})
