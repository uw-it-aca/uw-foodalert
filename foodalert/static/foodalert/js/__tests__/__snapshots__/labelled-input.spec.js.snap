// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Labelled Input renders correctly buttons 1`] = `
<div
  class="px-0 py-1 container-fluid"
  type="buttons"
>
  <div
    aria-labelledby="__BVID__17__BV_label_"
    class="labelled-textarea mb-0 b-form-group form-group"
    id="__BVID__17"
    role="group"
  >
    <label
      class="col-form-label pt-0"
      for="textarea-null"
      id="__BVID__17__BV_label_"
    >
      
            Label Text 
      <br />
    </label>
    <div
      class=""
    >
       
      <textarea
        class="form-control form-control-md"
        id="textarea-null"
        placeholder="Example Text"
        rows="1"
        wrap="soft"
      />
      <!---->
      <!---->
      <!---->
    </div>
  </div>
   
  <div
    aria-atomic="true"
    aria-live="polite"
    class="mt-1 alert alert-primary"
    placement="bottom"
    role="alert"
  >
    <!---->
     Warning Text 
  </div>
</div>
`;

exports[`Labelled Input renders correctly checkbox 1`] = `
<div
  class="px-0 py-1 container-fluid"
  type="checkbox"
>
  <div
    aria-labelledby="__BVID__7__BV_label_"
    class="labelled-textarea mb-0 b-form-group form-group"
    id="__BVID__7"
    role="group"
  >
    <label
      class="col-form-label pt-0"
      for="textarea-null"
      id="__BVID__7__BV_label_"
    >
      
            Label Text 
      <br />
    </label>
    <div
      class=""
    >
       
      <textarea
        class="form-control form-control-md"
        id="textarea-null"
        placeholder="Example Text"
        rows="1"
        wrap="soft"
      />
      <!---->
      <!---->
      <!---->
    </div>
  </div>
   
  <div
    aria-atomic="true"
    aria-live="polite"
    class="mt-1 alert alert-primary"
    placement="bottom"
    role="alert"
  >
    <!---->
     Warning Text 
  </div>
</div>
`;

exports[`Labelled Input renders correctly textarea 1`] = `
<div
  class="px-0 py-1 container-fluid"
>
  <div
    aria-labelledby="__BVID__2__BV_label_"
    class="labelled-textarea mb-0 b-form-group form-group"
    id="__BVID__2"
    role="group"
  >
    <label
      class="col-form-label pt-0"
      for="textarea-null"
      id="__BVID__2__BV_label_"
    >
      
            Label Text 
      <br />
    </label>
    <div
      class=""
    >
       
      <textarea
        class="form-control form-control-md"
        id="textarea-null"
        placeholder="Example Text"
        rows="1"
        wrap="soft"
      />
      <!---->
      <!---->
      <!---->
    </div>
  </div>
   
  <div
    aria-atomic="true"
    aria-live="polite"
    class="mt-1 alert alert-primary"
    placement="bottom"
    role="alert"
  >
    <!---->
     Warning Text 
  </div>
</div>
`;

exports[`Labelled Input renders correctly time 1`] = `
<div
  class="px-0 py-1 container-fluid"
  type="time"
>
  <div
    aria-labelledby="__BVID__12__BV_label_"
    class="labelled-textarea mb-0 b-form-group form-group"
    id="__BVID__12"
    role="group"
  >
    <label
      class="col-form-label pt-0"
      for="textarea-null"
      id="__BVID__12__BV_label_"
    >
      
            Label Text 
      <br />
    </label>
    <div
      class=""
    >
       
      <textarea
        class="form-control form-control-md"
        id="textarea-null"
        placeholder="Example Text"
        rows="1"
        wrap="soft"
      />
      <!---->
      <!---->
      <!---->
    </div>
  </div>
   
  <div
    aria-atomic="true"
    aria-live="polite"
    class="mt-1 alert alert-primary"
    placement="bottom"
    role="alert"
  >
    <!---->
     Warning Text 
  </div>
</div>
`;
